### PR TITLE
CI: Add rst backtick checker

### PR DESCRIPTION
**Pull request by dsaxton**
Adding a pre-commit hook for detecting single backticks around code in RST files. Running for instance on the v1.2.0 whatsnew shows errors:
```
doc/source/whatsnew/v1.2.0.rst:120:- `Styler` now allows direct CSS class name addition to individual data cells (:issue:`36159`)
doc/source/whatsnew/v1.2.0.rst:225:- Performance improvements when creating DataFrame or Series with dtype `str` or :class:`StringDtype` from array with many string elements (:issue:`36304`, :issue:`36317`, :issue:`36325`, :issue:`36432`)
doc/source/whatsnew/v1.2.0.rst:230:- Performance improvement in :meth:`pd.to_datetime` with non-`ns` time unit for `float` `dtype` columns (:issue:`20445`)
doc/source/whatsnew/v1.2.0.rst:265:- Bug in :func:`date_range` was raising AmbiguousTimeError for valid input with `ambiguous=False` (:issue:`35297`)
doc/source/whatsnew/v1.2.0.rst:306:- Bug in :meth:`SeriesGroupBy.transform` now correctly handles missing values for `dropna=False` (:issue:`35014`)
doc/source/whatsnew/v1.2.0.rst:312:- Bug in :meth:`DataFrame.xs` when used with :class:`IndexSlice` raises ``TypeError`` with message `Expected label or tuple of labels` (:issue:`35301`)
doc/source/whatsnew/v1.2.0.rst:320:- In :meth:`read_csv` `float_precision='round_trip'` now handles `decimal` and `thousands` parameters (:issue:`35365`)
doc/source/whatsnew/v1.2.0.rst:322:- :meth:`to_csv` passes compression arguments for `'gzip'` always to `gzip.GzipFile` (:issue:`28103`)
doc/source/whatsnew/v1.2.0.rst:324:- :meth:`to_csv` and :meth:`read_csv` did not honor `compression` and `encoding` for path-like objects that are internally converted to file-like objects (:issue:`35677`, :issue:`26124`, and :issue:`32392`)
doc/source/whatsnew/v1.2.0.rst:327:- Bug in :meth:`read_csv` with `engine='python'` truncating data if multiple items present in first row and first element started with BOM (:issue:`36343`)
doc/source/whatsnew/v1.2.0.rst:328:- Removed ``private_key`` and ``verbose`` from :func:`read_gbq` as they are no longer supported in `pandas-gbq` (:issue:`34654` :issue:`30200`)
```

~This is going to fail for now so need to figure out how to make it pass.~

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -30,3 +30,37 @@ repos:
     hooks:
     -   id: pyupgrade
         args: [--py37-plus]
+-   repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.6.0
+    hooks:
+      - id: rst-backticks
+        exclude: (?x)(
+          text\.rst|
```
Comment (dsaxton) [inline .pre-commit-config.yaml:38]: As follow up would be good to gradually delete these and fix the excluded files

Comment (jbrockmendel) [reply .pre-commit-config.yaml:38]: Can you add a comment to that effect (uh, assuming yaml supports comments)

File: doc/source/whatsnew/v1.2.0.rst [hunk 2]
```diff
@@ -303,29 +303,29 @@ Indexing
...
+- :meth:`to_csv` and :meth:`read_csv` did not honor ``compression`` and ``encoding`` for path-like objects that are internally converted to file-like objects (:issue:`35677`, :issue:`26124`, and :issue:`32392`)
 - :meth:`to_picke` and :meth:`read_pickle` did not support compression for file-objects (:issue:`26237`, :issue:`29054`, and :issue:`29570`)
 - Bug in :func:`LongTableBuilder.middle_separator` was duplicating LaTeX longtable entires in the List of Tables of a LaTeX document (:issue:`34360`)
-- Bug in :meth:`read_csv` with `engine='python'` truncating data if multiple items present in first row and first element started with BOM (:issue:`36343`)
-- Removed ``private_key`` and ``verbose`` from :func:`read_gbq` as they are no longer supported in `pandas-gbq` (:issue:`34654` :issue:`30200`)
+- Bug in :meth:`read_csv` with ``engine='python'`` truncating data if multiple items present in first row and first element started with BOM (:issue:`36343`)
+- Removed ``private_key`` and ``verbose`` from :func:`read_gbq` as they are no longer supported in ``pandas-gbq`` (:issue:`34654` :issue:`30200`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.2.0.rst:328]: comma between issue references?

**Comment by jbrockmendel** [comment]
nitpicks, generally looks good

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dsaxton) [reply .pre-commit-config.yaml:39]: Added

File: doc/source/whatsnew/v1.2.0.rst [hunk 3]
```diff
@@ -222,12 +222,12 @@ Deprecations
...
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-- Performance improvements when creating DataFrame or Series with dtype `str` or :class:`StringDtype` from array with many string elements (:issue:`36304`, :issue:`36317`, :issue:`36325`, :issue:`36432`)
+- Performance improvements when creating DataFrame or Series with dtype ``str`` or :class:`StringDtype` from array with many string elements (:issue:`36304`, :issue:`36317`, :issue:`36325`, :issue:`36432`)
 - Performance improvement in :meth:`GroupBy.agg` with the ``numba`` engine (:issue:`35759`)
 - Performance improvements when creating :meth:`pd.Series.map` from a huge dictionary (:issue:`34717`)
 - Performance improvement in :meth:`GroupBy.transform` with the ``numba`` engine (:issue:`36240`)
 - ``Styler`` uuid method altered to compress data transmission over web whilst maintaining reasonably low table collision probability (:issue:`36345`)
-- Performance improvement in :meth:`pd.to_datetime` with non-`ns` time unit for `float` `dtype` columns (:issue:`20445`)
+- Performance improvement in :meth:`pd.to_datetime` with non-``ns`` time unit for ``float`` ``dtype`` columns (:issue:`20445`)
```
Comment (WillAyd) [inline doc/source/whatsnew/v1.2.0.rst:230]:
```suggestion
- Performance improvement in :meth:`pd.to_datetime` with non-`ns` time unit for ``float`` ``dtype`` columns (:issue:`20445`)
```

Somewhat pedantic but this seems incorrect to treat as a literal

Comment (dsaxton) [reply doc/source/whatsnew/v1.2.0.rst:230]: Maybe no backticks at all? I think this is going to fail with single

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jreback) [reply .pre-commit-config.yaml:39]: also can you create an issue for this (with checkboxes)

File: doc/source/whatsnew/v1.2.0.rst (hunk 3 above)
Comment (jreback) [reply doc/source/whatsnew/v1.2.0.rst:230]: yeah this should be double or nothing

**Review by jreback — COMMENTED** [review]
lgtm. pls merge when ready

**Pull request merged**